### PR TITLE
Allow using custom config in `babel-node --eval`

### LINKED ISSUE #11877: [babel-node] babel-node won't use --config-file when using -e codeToEval
## Bug Report

- [x] I would like to work on a fix!

**Current behavior**

Given a configuration file of:

```js
// .test-babelrc.js
module.exports = {
  presets: ["@babel/preset-env"],
};
```

And a command line of:

```bash
yarn babel-node --config-file ./.test-babelrc.js -e "const a = 1"
```

It will error with: "SyntaxError: /[redacted path]/[eval]: Only `var` variables are supported in the REPL". Because it's actually not using the configuration file.

The configuration file is not used to eval the code. You can even give the command line a bad configuration file path or mess up the inside syntax of the configuration: it does not matter, it won't be used.

If you rename `.test-babelrc.js` to `.babelrc.js` and change command to:

```bash
yarn babel-node -e "const a = 1"
```

Then it works, because somehow now the default babel configuration file location is used.
Also, if you use:

```bash
./node_modules/.bin/babel-node --config-file ./.test-babelrc.js ./test.js
```

Then it works, given a test.js file of:

```js
// test.js
const a = 1;
console.log(a);
```

PS: I **need** to use a different default configuration file path (.test-babelrc.js) in the context of my project.

**Input Code**

I created a repository to reproduce the bug here: https://github.com/vvo/test-babel-node-configuration-file

**Expected behavior**
The custom configuration file path should be used when combined with the eval code.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `babel.config.js`

```js
module.exports = {
  presets: ["@babel/preset-env"],
};
```

**Environment**

```
  System:
    OS: macOS 10.15.5
  Binaries:
    Node: 12.18.3 - ~/.nvm/versions/node/v12.18.3/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.14.6 - ~/.nvm/versions/node/v12.18.3/bin/npm
  npmPackages:
    @babel/core: 7.10.5 => 7.10.5
    @babel/node: 7.10.5 => 7.10.5
    @babel/preset-env: 7.10.4 => 7.10.4
```
- Monorepo: n/a
- How you are using Babel: babel-node

**Possible Solution**
n/a

**Additional context**
n/a

### LINKED ISSUE #16675: [Bug]: Script evaluation using babel-node’s ‘--eval’ doesn’t apply default config file
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/register or @babel/node

### Input code

```js
console.log("foo");
```

### Configuration file name

babel.config.js

### Configuration

```js
module.exports = {
  plugins: [
    [ "search-and-replace", { rules: [ { search: "foo", replace: "bar" } ] } ],
  ],
};
```

### Current and expected behavior

Using a file ‘test.js’ with the input code given above, and a file ‘babel.config.js’ with the given configuration which uses a plugin to replace `"foo"` by `"bar"`, the following has the expected result:

```
$ npx babel-node test.js
bar
```

But when using `--eval` to pass the input code, the result is unexpected, the config file does not seem to be applied:

```
$ npx babel-node --eval 'console.log("foo");'
foo
```

### Environment

```
  System:
    OS: macOS 13.6.7
  Binaries:
    Node: 20.16.0 - ~/.nvm/versions/node/v20.16.0/bin/node
    npm: 10.8.1 - ~/.nvm/versions/node/v20.16.0/bin/npm
  npmPackages:
    @babel/node: ^7.24.8 => 7.24.8 
    babel-plugin-search-and-replace: ^1.1.1 => 1.1.1 
```

### Possible solution

_No response_

### Additional context

Issue #11877, which is about `--config-file` not being applied, is similar but differs in that it says that a default config file did work.

### PR DESCRIPTION
**Pull request by slatereax**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #11877, fixes #16675
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Transform code for `babel-node` eval does not use the options that were passed in the cli flags. Also, `replPlugin`, which is used for code transformation, only allows `var` variables, although modern versions of node repl support `let` and `const`.

Based on the history of this file, it appears that this code was written in 2017 when node did not support any variables other than var, but there is no need for this check now. If anyone needs to use babel-node with node < 6.0.0, they can pass their configuration file through --config-file.

### CODE REVIEW

File: packages/babel-node/src/_babel-node.ts [hunk 1]
```diff
@@ -151,8 +143,8 @@ const _eval = function (code: string, filename: string) {
 
   code = babel.transformSync(code, {
     filename: filename,
-    presets: opts.presets,
-    plugins: (opts.plugins || []).concat([replPlugin]),
+    ...babelOptions,
+    plugins: (opts.babelOptions || []).concat([replPlugin]),
```
Comment (JLHwung) [inline packages/babel-node/src/_babel-node.ts:147]:
```suggestion
    plugins: (opts.plugins || []).concat([replPlugin]),
```
`babelOptions` is not a valid option name anyway.

Comment (slatereax) [reply packages/babel-node/src/_babel-node.ts]: Oh sorry, my mistake, just wanted to use babelOptions.plugins here

**Review by JLHwung — APPROVED** [review]
Thanks.

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

**Comment by JLHwung** [comment]
@slatereax 
I realize that this PR does not add new tests, could you add two new tests to `packages/babel-node/test/fixtures/cli` ? One for #11877 and another one for #16675. You can check that folder to see how tests are constructed.

**Comment by nicolo-ribaudo** [comment]
@JLHwung I added two tests, please take a look!

**Comment by JLHwung** [comment]
> @JLHwung I added two tests, please take a look!

Thanks, in #11877 we do load configs but they are not passed to the eval transform so the new tests are actually passing on the main branch. Can you add a simple plugin to manipulate the input source? So that we are sure the resolved config is applied.

**Comment by nicolo-ribaudo** [comment]
Done!

**Pull request merged**